cmake/tpl/SundialsCaliper.cmake

```diff
@@ -86,16 +70,17 @@ if(CALIPER_FOUND AND (NOT CALIPER_WORKS))
     "  return 0;\n"
     "}\n")
 
-  # To ensure we do not use stuff from the previous attempts, we must remove the
-  # CMakeFiles directory.
-  file(REMOVE_RECURSE ${CALIPER_TEST_DIR}/CMakeFiles)
-
-  # Attempt to build and link the "ltest" executable
+  # Attempt to build and link executable with caliper
   try_compile(
     COMPILE_OK ${CALIPER_TEST_DIR}
-    ${CALIPER_TEST_DIR} ltest
+    ${CALIPER_TEST_DIR}/ltest.${_ext}
+    LINK_LIBRARIES caliper
     OUTPUT_VARIABLE COMPILE_OUTPUT)
 
+  # To ensure we do not use stuff from the previous attempts, we must remove the
+  # CMakeFiles directory.
+  file(REMOVE_RECURSE ${CALIPER_TEST_DIR}/CMakeFiles)
+
```

Comment:
`try_compile` cleans up after itself 
```suggestion
```

Reply:
To be clear, `try_compile` cleans up after itself only when using it in the "source file" mode (like we are now). This line was leftover from before where we used the "project" mode.